frontend/public/components/modals/remove-idp-modal.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { useTranslation, Trans } from 'react-i18next';
+import { k8sPatchResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  ModalWrapper,
+} from '@console/internal/components/factory/modal';
+import { ModalComponent } from '@console/dynamic-plugin-sdk/src/app/modal-support/ModalProvider';
+import { HandlePromiseProps, withHandlePromise } from '@console/internal/components/utils';
+import { K8sModel, OAuthKind } from '@console/internal/module/k8s';
+import { YellowExclamationTriangleIcon } from '@console/shared';
+
+const RemoveIdentityProviderModalComponent = withHandlePromise(
+  ({
+    obj,
+    model,
+    index,
+    name,
+    type,
+    inProgress,
+    errorMessage,
+    cancel,
+    close,
+    handlePromise,
+  }: RemoveIdentityProviderModalProps) => {
+    const handleSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
```

Comment:
Need to use React.useCallback. This callback is dependent on several props.

Reply:
Yes, the callback is dependent on several props. But the function is only used locally, it is not passed as a prop or used in dependency arrays, and does not directly cause unnecessary re-renders of child components or used in dependency arrays of hooks. Athough the function instance is recreated on re-render. Is there a need to memoize it?